core/cpu.c

```diff
@@ -58,6 +59,17 @@ static int cpu_nx_enable()
     return effer & 0x800;
 }
 
+bool cpu_has_feature(uint32_t feature)
+{
+    static cpuid_cache_t cache = {
+        .initialized = 0
+    };
+    if (!cache.initialized) {
+        cpuid_host_init(&cache);
```

Comment:
nit: Is there still a race condition, when two host CPUs both see the cache as uninitialized and try to fill it at the same time? This may not cause a problem, since they only write to the cache and should write the same data. Still, it would be nice if only one CPU performed the initialization. C++11 actually guarantees that static local initializers are only called once:

https://en.cppreference.com/w/cpp/language/storage_duration#Static_local_variables

I'm not sure if we can take advantage of that, because this is C code, and our compilers may not support or be configured to use C++11.

A simple fix would be to call `cpu_has_feature()` with an arbitrary feature very early, e.g. before this call:

https://github.com/intel/haxm/blob/2c0ac6c203e30f61f09184f8d36d6d2b5396234d/core/hax.c#L543

so as to force CPU cache initialization in a non-`smp_call_function()` context.